marque and map section

Kaarthik-07 opened the pull request:
Marquee and Carosuel done
marquee functions :
hover on stop
hover on opacity reduce
responsive done

Map card button
Functions:
on hover map position changes
Added button for two cards section
Responsive design done
Smooth animation done for both sections

Dependency: to be installed
yarn add react-fast-marquee
yarn add framer-motion

MKMukeshkannan commented on package.json:
```diff
@@ -27,5 +30,9 @@
     "tailwindcss": "^3.3.0",
     "typescript": "^5"
-  }
+  },
```
- Remove the author, license, repository and main keys
- Kindly rename the images with proper name.

MKMukeshkannan commented on src/components/CardButtonn.tsx:
```diff
@@ -0,0 +1,62 @@
+  return (
+    <>
+      
+          <div className="flex justify-evenly">
```
- replace div with section of better code semantics

MKMukeshkannan commented on src/components/Marquee.tsx:
```diff
@@ -0,0 +1,115 @@
+const Images = [
+  {
+    id: 1,
+    src :'https://raw.githubusercontent.com/jesvijonathan/Takshashila/master/static/sponsors/Group%20298%20(1).webp',
```
- do not use external images, use images in the repository

MKMukeshkannan commented on src/components/NavBar.tsx:
```diff
@@ -15,7 +15,7 @@ const NavBar = () => {
 
   return (
-    <nav className="flex justify-between items-center h-16 fixed z-50 w-full p-8 pt-16 md:p-20">
+    <nav className="flex justify-between items-center h-16 fixed z-50 w-full p-8 pt-16 md:p-20 res">
```
- try to use responsive breakpoint given in the tailwind, try to avoid writing custom breakpoints

MKMukeshkannan commented on src/components/ToggleButton.tsx:
```diff
@@ -0,0 +1,47 @@
+
+            {/* <div className="flex justify-center items-center h-screen"> */}
+        <div className="defaultStyle" style={{}}>             
+                 <div className="" style={{margin:'2%'}}>
```
- don't use explicit inline styling, try to stay with tailwind classes

MKMukeshkannan commented on src/components/app.css:
```diff
@@ -0,0 +1,30 @@
+.defaultStyle{
```
- implement all these style with tailwind.

MKMukeshkannan reviewed (changes requested):
- Kindly remove the package-lock.json file
- Use section instead of div for better code semantics
- Rename images
- Try to stay within tailwindcss and limit yourself from using inline styles and global css

Kaarthik-07 commented:
** Enhance Responsiveness and Update Marquee Animation**

### Changes Introduced
This pull request introduces several updates aimed at improving the application's responsiveness and enhancing visual animations. Here are the key changes:

- **Responsive Design Adjustments**: Implemented Tailwind CSS responsive classes to improve the layout on small (`sm`), medium (`md`), and large (`lg`) devices. This ensures that components like the map card section adapt smoothly across different screen sizes.
  
- **Marquee Animation**: Enhanced the marquee section with smoother transitions and animations, providing a more dynamic user experience.
  
- **Image Naming**: Renamed image assets for better consistency and to follow naming conventions, which will assist in maintainability and scalability.

- **Padding Reduction**: Reduced padding in various components to allow for more screen real estate and a cleaner layout, particularly on mobile devices.


- **Small Devices**:
- 

![Screenshot 2024-02-04 185148](https://github.com/cittakshashila/frontend/assets/132292791/4bf704a7-8a7b-4abc-b4b2-f9f85114f7b7)
![Screenshot 2024-02-04 185204](https://github.com/cittakshashila/frontend/assets/132292791/82b00c1b-1714-467b-8a65-7cf574196269)

**Medium Devices
![Screenshot 2024-02-04 185217](https://github.com/cittakshashila/frontend/assets/132292791/10fdbbf3-e72a-4426-958a-4ad655af19ee)
![Screenshot 2024-02-04 185225](https://github.com/cittakshashila/frontend/assets/132292791/8594cc2b-db0e-4940-990c-e804eec6261f)

-**Large Devices**:
![Screenshot 2024-02-04 185240](https://github.com/cittakshashila/frontend/assets/132292791/4145d313-55a7-464d-94ef-2c3d969e7307)
![Screenshot 2024-02-04 185251](https://github.com/cittakshashila/frontend/assets/132292791/d4fd3cd6-856c-44fd-98f8-6e985dd5689e)
